### PR TITLE
Minio and S3 compatible storage fixes

### PR DESCRIPTION
**Pull request by iamrz1**
- [x] fixes for test fail
- [x] storage agnostic endpoint (any of s3,mioo/other compatible storage)
- [x] region for compatible storages

### CODE REVIEW

**Comment by iamrz1** [comment]
Fixes endpoint and region issue in https://github.com/kubedb/project/issues/568

**Comment by iamrz1** [comment]
Fixes minio test fail (with and without ca-cert) in https://github.com/kubedb/project/issues/623

**Comment by tamalsaha** [comment]
I think we need to cut new version of the postgres images for this fix.

**Comment by the-redback** [comment]
@iamrz1 , can you please fix postgres docker image build? It's failing because some of the scripts are not available anymore.

File: test/e2e/framework/minio.go [hunk 1]
```diff
@@ -586,23 +586,40 @@ func (fi *Invocation) DeleteMinioServer() (err error) {
...
+	err = fi.DeleteDeploymentForMinioServer(mdeploy.ObjectMeta)
 	return err
 }
 
 func (f *Framework) DeleteSecretForMinioServer(meta metav1.ObjectMeta) error {
-	return f.kubeClient.CoreV1().Secrets(meta.Namespace).Delete(meta.Name, deleteInForeground())
+	_, err := f.kubeClient.CoreV1().Secrets(meta.Namespace).Get(meta.Name, metav1.GetOptions{})
+	if err == nil {
+		return f.kubeClient.CoreV1().Secrets(meta.Namespace).Delete(meta.Name, deleteInForeground())
+	}
```
Comment (the-redback) [inline test/e2e/framework/minio.go]: We should not do it. If you need, handle the error from where this function is getting called.

File: test/e2e/framework/minio.go [hunk 2]
```diff
@@ -586,23 +586,40 @@ func (fi *Invocation) DeleteMinioServer() (err error) {
...
+	return nil
 }
 
 func (f *Framework) DeletePVCForMinioServer(meta metav1.ObjectMeta) error {
-	return f.kubeClient.CoreV1().PersistentVolumeClaims(meta.Namespace).Delete(meta.Name, deleteInForeground())
-}
+	_, err := f.kubeClient.CoreV1().PersistentVolumeClaims(meta.Namespace).Get(meta.Name, metav1.GetOptions{})
```
Comment (the-redback) [inline test/e2e/framework/minio.go]: same goes here

File: test/e2e/framework/minio.go [hunk 3]
```diff
@@ -586,23 +586,40 @@ func (fi *Invocation) DeleteMinioServer() (err error) {
...
+	}
+	return nil
 
-func (f *Framework) DeleteDeploymentForMinioServer(meta metav1.ObjectMeta) error {
-	return f.kubeClient.AppsV1().Deployments(meta.Namespace).Delete(meta.Name, deleteInBackground())
 }
 
 func (f *Framework) DeleteServiceForMinioServer(meta metav1.ObjectMeta) error {
-	return f.kubeClient.CoreV1().Services(meta.Namespace).Delete(meta.Name, deleteInForeground())
+	_, err := f.kubeClient.CoreV1().Services(meta.Namespace).Get(meta.Name, metav1.GetOptions{})
```
Comment (the-redback) [inline test/e2e/framework/minio.go]: same goes here

File: test/e2e/postgres_test.go [hunk 4]
```diff
@@ -20,8 +17,10 @@ import (
 	"kubedb.dev/apimachinery/client/clientset/versioned/typed/kubedb/v1alpha1/util"
 	"kubedb.dev/postgres/test/e2e/framework"
 	"kubedb.dev/postgres/test/e2e/matcher"
+	"os"
```
Comment (the-redback) [inline test/e2e/postgres_test.go]: run `make`

File: test/e2e/postgres_test.go [hunk 5]
```diff
@@ -1698,21 +1698,25 @@ var _ = Describe("Postgres", func() {
 					By("By definition, WAL files can not be initialized from EmptyDir. Skipping Test.")
 				})
 
-				Context("With PVC as Archive backend", func() {
-					var archivePVC *core.PersistentVolumeClaim
+				FContext("With PVC as Archive backend", func() {
```
Comment (the-redback) [inline test/e2e/postgres_test.go]: undo Focused test

**Pull request merged**